account_payment_line/tests/test_account_payment_line.py

```diff
@@ -0,0 +1,987 @@
+# Copyright 2022 ForgeFlow, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+import json
+
+from odoo import fields
+from odoo.exceptions import ValidationError
+from odoo.tests import Form, tagged
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+
+@tagged("post_install", "-at_install")
+class TestAccountPaymentLines(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.test_product = cls.env["product.product"].create(
+            {
+                "name": "test_product",
+                "type": "service",
+            }
+        )
+        cls.customer = cls.env["res.partner"].create(
+            {
+                "name": "test_customer",
+            }
+        )
+        cls.customer2 = cls.env["res.partner"].create(
+            {
+                "name": "test_customer",
+            }
+        )
+        cls.supplier = cls.env["res.partner"].create(
+            {
+                "name": "test_vendor",
+            }
+        )
+        cls.supplier2 = cls.env["res.partner"].create(
+            {
+                "name": "test_vendor",
+            }
+        )
+        cls.currency_2x = cls.env["res.currency"].create(
+            {
+                "name": "2X",  # Foreign currency, 2 time
+                "symbol": "X",
+                "rate_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": fields.Date.today(),
+                            "rate": cls.env.company.currency_id.rate * 2,
+                        },
+                    )
+                ],
```

Comment:
Please use `context_today` to prevent issues in countries where the time zone is negative.

```suggestion
                "rate_ids": [
                    Command.create(
                        {
                            "name": fields.Date.context_today(cls.env.user),
                            "rate": cls.env.company.currency_id.rate * 2,
                        },
                    )
                ],
```